app/javascript/admin/controllers/creator_settings_controller.js

```diff
@@ -67,4 +67,53 @@ export class CreatorSettingsController extends Controller {
 
     reader.readAsDataURL(firstFile);
   }
+
+  /**
+   * Updates ths branding/colors on the Creator Settings Page.
+   *
+   * @param {Event} event
+   */
+  updateBranding(event) {
+    const color = event.target.value;
+
+    document.documentElement.style.setProperty('--accent-brand', color);
+
+    // responsible for the hover effect over the button.
+    // We need to recalculate this in javascript as it's currently being calculated in ruby
+    document.documentElement.style.setProperty(
+      '--accent-brand-darker',
+      this.updatedBrightness(color, 0.85),
+    );
+  }
+
+  updatedBrightness(color, amount = 1) {
+    const rgbObj = this.hexToRgb(color);
+    Object.keys(rgbObj).forEach((key) => {
+      rgbObj[key] = Math.round(rgbObj[key] * amount);
+    });
+
+    return this.rgbToHex(rgbObj['r'], rgbObj['g'], rgbObj['b']);
+  }
+
+  hexToRgb(hex) {
+    const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
```

Comment:
(not blocking) maybe it could be useful to add a comment about what's happening in this regex

Reply:
I added some descriptions for the functions in the utility classes to help with this ✅